server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -2026,6 +2026,7 @@ private static String previousWriterMessage(@Nullable Tuple<Long, String> previo
     private void markRepoCorrupted(long corruptedGeneration, Exception originalException, ActionListener<Void> listener) {
         assert corruptedGeneration != RepositoryData.UNKNOWN_REPO_GEN;
         assert bestEffortConsistency == false;
+        logger.warn(() -> "Marking repository [" + metadata.name() + "] as corrupted", originalException);
```

Comment:
nit: the message-supplier form is unnecessary here:

```suggestion
        logger.warn("Marking repository [" + metadata.name() + "] as corrupted", originalException);
```

Reply:
I was assuming we prefer it to avoid string concatenation (with repository name in this case). Is it because of warn?